frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
+		///
+		/// Subbounty must be in active state, for
+		/// processing the call. and state of subbounty is
+		/// moved to PendingPayout on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		/// - `beneficiary`: Beneficiary account.
+		#[weight = 10_000]
+		fn award_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+			beneficiary: <T::Lookup as StaticLookup>::Source
+		) {
+			let signer = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			// Ensure parent bounty is Active
+			let ( master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Ensure subbounty is in expected state
+			SubBounties::<T>::try_mutate_exists(bounty_id, subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure Subbounty is in active state
+				match &subbounty.status {
+					SubBountyStatus::Active {
+						subcurator,
+						..
+					} => {
+						// TODO :: Have to recheck
+						// Can master curator award subbounty
+						// When subcurator is assigned ?
+						ensure!(
+							signer == *subcurator || signer == master_curator,
+							Error::<T>::RequireCurator,
+						);
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				}
+				// Move the subbounty state to Pending payout.
+				// TODO :: Have to recheck
+				// Do we need trait config for subbounty,
+				// instead of using T::BountyDepositPayoutDelay ?
+				subbounty.status = SubBountyStatus::PendingPayout {
+					subcurator: signer,
+					beneficiary: beneficiary.clone(),
+					unlock_at: system::Module::<T>::block_number() +
+						T::BountyDepositPayoutDelay::get(),
+				};
+				Ok(())
+			})?;
+			// Trigger the event SubBountyAwarded
+			Self::deposit_event(Event::<T>::SubBountyAwarded(bounty_id, subbounty_id, beneficiary));
+		}
+
+		/// Claim the payout from an awarded subbounty after payout delay.
+		///
+		/// The dispatch origin for this call may be any signed origin.
+		///
+		/// Call works independent of parent bounty state,
+		/// No need for parent bounty must be in active state.
+		///
+		/// Beneficiary is paid out with with agreed bounty value.
+		/// SubCurator fee is paid & bond deposit is unreserved.
+		///
+		/// Subbounty must be in PendingPayout state, for
+		/// processing the call. and instance of subbounty is
+		/// deallocated from DB on successful call completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = 10_000]
+		fn claim_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let _ = ensure_signed(origin)?;
+
+			// TODO :: Have to recheck
+			// ignoring the requirement of parent bounty should be active
```

Comment:
I think that's fine. parent curator can always close subbounty if needed

Reply:
This is about "claim_subbounty()",
Right now implementation is like, any signed origin can execute the call and make the claiming process ( transfer fund to assigned beneficiary & subcurator ). And no need for parent bounty to be active. Is this sequence flow OK ?